modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -737,6 +737,16 @@ def _reset(df, *axis_lengths, partition_idx):
             )
         return new_self
 
+    def maybe_reset_index_to_positional(self):
+        if self.index.equals(pandas.RangeIndex(len(self.index))):
+            return self
+        return self.reset_index(drop=True)
+
+    def maybe_reset_columns_to_positional(self):
```

Comment:
We could have also added the `axis` parameter to the existing `qc.reset_index` method in order not to produce new methods to already overloaded query compiler class. What do you think?